protected/humhub/components/UrlManager.php

```diff
@@ -43,4 +44,12 @@ public function createUrl($params)
 
         return parent::createUrl($params);
     }
+
+    /**
+     * @inheritdoc
+     */
+    public function getHostInfo()
+    {
+        return Yii::$app->settings->get('baseUrl') ?: parent::getHostInfo();
```

Comment:
Please note that `baseUrl` can also something like `https://localhost/folder/humhub`.

https://www.yiiframework.com/doc/api/2.0/yii-web-request#getHostInfo()-detail

`getHostInfo` should only return `https://localhost`

Reply:
@luke- Please see my test when `getHostInfo()` returns `http://localhost`:

![folder](https://user-images.githubusercontent.com/6995524/190434743-8143528f-76be-41ff-878a-047b4b56b878.png)

Reply:
This is strange, in my test cases I always got back a correct absolute URL when I return `http://localhost` there for example.

My Tests:
- With and without enabled Pretty URL in Root Folder
- With and without enabled Pretty URL in '/humhub` subfolder

As webserver I'm using NGINX. I added my WebServer Config for the subfolder setup here:
https://community.humhub.com/s/installation-and-setup/wiki/WebServer+Configurations

Maybe it is your WebServer Config?

Reply:
@luke- Sorry, my previous test was not completely correct, because Base URL was defined as `http://localhost/humhub` but the request a password was done from alias `http://humhub.yb`.

![test-hosts](https://user-images.githubusercontent.com/6995524/190571534-1c93d905-9eda-4687-8321-58b2c63a0309.png)

I.e. yesterday I tested only the case `2a`.

So if we return only host as `http://localhost` then case `2a` will be broken, however this solution will fixes the case `1b`.
I think to find a solution for all cases.

Reply:
@yurabakhtin Your case `2a` is very special, you assume here that HumHub can run in parallel in the root folder as well as in a subdirectory. I would just assume that the domain can change.

Reply:
FYI: here is how we use the base url in the console part: https://github.com/humhub/humhub/blob/master/protected/humhub/components/console/Application.php#L80-L90

Reply:
@luke- I wanted to post this comment 5 hours ago, but it seems I just forgot to press "Comment", and it was stored here:

---

Maybe it is not correctly to use a site from alias like `http://humhub.yb` when base URL is defined as `http://localhost/humhub`, but I have implemented a solution here https://github.com/humhub/humhub/pull/5864/commits/ae1886fbfa9a72389f8238a06bf2a0d2303c33ae

my tests for the code `Url::to(['/user/password-recovery/reset'], true)`:

![fixed_host](https://user-images.githubusercontent.com/6995524/190588197-51664f33-1a23-4abf-93ca-8b86954c8c7e.png)